contracts/Storage.sol

```diff
@@ -79,8 +82,20 @@ contract Storage is Collateral, Marketplace {
   function markProofAsMissing(bytes32 slotId, uint256 period) public {
     require(!isSlotCancelled(slotId), "Request was cancelled");
     _markProofAsMissing(slotId, period);
+    address host = _host(slotId);
     if (_missed(slotId) % slashMisses == 0) {
-      _slash(_host(slotId), slashPercentage);
+
+      uint256 slashAmount = _slashAmount(host, slashPercentage);
+      if (balanceOf(host) - slashAmount < minCollateralThreshold) {
```

Comment:
Yes, it's likely simpler. However, the gas costs would probably be higher as there would be a slash immediately followed by a slot being freed. We can avoid the slashing with a bit of logic. Happy to change it over if it's too complex.

Reply:
I would actually expect the host to both be slashed (as punishment for missing a proof) and kicked from the slot (because they missed too many proofs). At least as long as we have a global collateral, instead of a collateral per slot.